packages/tslint-config/src/rules/blueprintClassesConstantsRule.ts

```diff
@@ -29,15 +28,144 @@ export class Rule extends Lint.Rules.AbstractRule {
+    }
+
+    if (utils.isStringLiteral(node)) {
+        let stringWithoutPtClasses = node.getText();
+        ptClassStrings.forEach(cssClass => {
+            stringWithoutPtClasses = stringWithoutPtClasses.replace(cssClass, "");
+        });
+        const templateStrings = ptClassStrings.map(n => `\${${convertPtClassName(n)}}`).join(" ");
+        // get rid of leading/ending quotes and any leading/trailing white space
+        stringWithoutPtClasses = stringWithoutPtClasses.slice(1, -1).trim();
+        if (stringWithoutPtClasses.length > 0) {
+            const replacement = `\`${templateStrings} ${stringWithoutPtClasses}\``;
+            replacements.push(
+                new Lint.Replacement(node.getStart(), node.getWidth(), wrapForParent(replacement, node, node.parent))
+            );
+        } else {
+            if (ptClassStrings.length === 1) {
+                const replacement = convertPtClassName(ptClassStrings[0]);
+                replacements.push(
+                    new Lint.Replacement(
+                        node.getStart(),
+                        node.getWidth(),
+                        wrapForParent(replacement, node, node.parent)
+                    )
+                );
+            } else {
+                const replacement = `\`${templateStrings}\``;
+                replacements.push(
+                    new Lint.Replacement(
+                        node.getStart(),
+                        node.getWidth(),
+                        wrapForParent(replacement, node, node.parent)
+                    )
+                );
+            }
+        }
+    } else if (utils.isTemplateExpression(node) || utils.isNoSubstitutionTemplateLiteral(node)) {
+        let replacementText = node.getText();
+        ptClassStrings.forEach(classString => {
+            const classReplacement = `\${${convertPtClassName(classString)}}`;
+            replacementText = replacementText.replace(classString, classReplacement);
+        });
+        replacements.push(new Lint.Replacement(node.getStart(), node.getWidth(), replacementText));
+    }
+    return replacements;
+}
+
+function wrapForParent(statement: string, node: ts.Node, parentNode: ts.Node | undefined) {
+    if (parentNode) {
+        if (utils.isJsxAttribute(parentNode)) {
+            return `{${statement}}`;
+        } else if (utils.isExpressionStatement(parentNode)) {
+            return `[${statement}]`;
+        } else if (utils.isPropertyAssignment(parentNode)) {
+            if (parentNode.getChildAt(0) === node) {
+                // If we're changing the key, we need to wrap it. Else, we're changing a value,
+                // and there's no need to wrap
+                return `[${statement}]`;
             }
+            return statement;
+        } else {
+            return statement;
         }
-        return ts.forEachChild(node, cb);
-    });
+    } else {
+        return statement;
```

Comment:
can you refactor nicely so there's only one instance of `return statement;`?

Reply:
down to two -- since i chose to use the first check as a type gate and not repeat an undefined check in the subsequent statements. If you think it's more readable with these combined, I'm happy to make a change again.

Reply:
ok this is nice and flat 👍